requestmanager/asyncloader/asyncloader.go

```diff
@@ -37,11 +34,10 @@ type Allocator interface {
 // AsyncLoader manages loading links asynchronously in as new responses
 // come in from the network
 type AsyncLoader struct {
-	ctx              context.Context
-	cancel           context.CancelFunc
-	incomingMessages chan loaderMessage
-	outgoingMessages chan loaderMessage
+	ctx    context.Context
+	cancel context.CancelFunc
 
+	stateLk           sync.Mutex
```

Comment:
I'd explicitly document what fields this mutex protects, even if that's just "all fields below".

Reply:
resolved